Marlin/src/module/temperature.h

```diff
@@ -506,8 +510,8 @@ class Temperature {
       static millis_t preheat_end_time[HOTENDS];
     #endif
 
-    #if HAS_AUTO_FAN
-      static millis_t next_auto_fan_check_ms;
+    #if HAS_AUTO_FAN || HAS_FANCHECK
```

Comment:
What does `HAS_FANCHECK` apply to, if not `HAS_AUTO_FAN` ?

We don't necessarily have to link the code that checks the hotend temperature and turns the fans on/off with the code that reads the auto fan tachometers. In this way the extruder autofans can check the temperature at one interval (e.g., 2500 ms) to manage turning them on/off, and the fan tachometer reading can occur at whatever interval is best for the relevance of the tachometer value.

Reply:
Not sure I can correctly understand the meaning of

> What does HAS_FANCHECK apply to, if not HAS_AUTO_FAN ?

I try to give an answer and I hope I got what you meant.
HAS_FANCHECK has to check tacho, always. When HAS_AUTOFAN I need to deal with it because fans may be turned off.
On/off is handled in `update_autofans` and speed is saved to `autofan_speed[realFan]` that's why 'check' call is done "after" it (just restored on my repo).

Tacho check time is lowered to 1000ms to have better control on "timeout". Since tacho error detection will stop heating to prevent heat creep time this time is hard predictable. Prusa uses 15s but it seems to big to me.

Reply:
Also note I think I have to restore TERN_...whathere was just a placeholder, I think correct code should be "kill" because I can't find out a correct way to pause printer when no lcd is installed.